src/ng/directive/ngEventDirs.js

```diff
@@ -64,10 +64,13 @@ forEach(
               var callback = function() {
                 fn(scope, {$event: event});
               };
-              if (forceAsyncEvents[eventName] && $rootScope.$$phase) {
+
+              if (!$rootScope.$$phase) {
+                scope.$apply(callback);
+              } else if (forceAsyncEvents[eventName]) {
                 scope.$evalAsync(callback);
               } else {
-                scope.$apply(callback);
+                callback();
```

Comment:
So, we are not wrapping this in `try { ... } catch (err) { $exceptionHandler(Err); }`?

Reply:
That should make the following more consistent...
```
dom.click();
otherStuff();
```
... when `dom.click()` invokes `ng-click` and does vs does-not throw. SGTM...